src/core/src/bootstrap/Bootstrapper.py

```diff
@@ -23,6 +23,7 @@
 from core.src.bootstrap.Constants import Constants
 from core.src.bootstrap.Container import Container
 from core.src.local_loggers.StdOutFileMirror import StdOutFileMirror
+from core.src.service_interfaces.TelemetryWriter import TelemetryWriter
 
 
```

Comment:
**Overarching comment for code as in this iteration:**
I understand the reason telemetry writer is only fully initialized in PatchAssessor is so that the error can be reported.

Conceptually and in the specific implementation however, this creates a tight coupling between those classes, and also relies on a specific assumption that the order of operation execution will remain the same way forever (assess then patch (then assess)) creating rigidity in the code and a subsequent fragility. [Academic comment since this doesn't apply to what is requested below: telemetry_writer should be idempotent vs. assessor avoiding a second call into it]

Consider this (or similar instead):::
Initialize telemetry writer within bootstrapper.
- Telemetry writer should internally store in a variable if it failed to initialize because the agent doesn't support it.
      - The way it's now causes tight coupling (see comment elsewhere).
- This failure to initialize can later be consumed in both the Assessor and Installation classes to report into status.
- The highest catch block in Core Main whenever it sees a *first* (not second!) assessment failure on an installation operation, reports into the installation block in status also (covers all first assessment failures).

How to initialize within bootstrapper
- See line 68 for deserialized environment_settings.
- Pull it out similar to that from environment_settings, pass it back out to __init__ as a third variable telemetry_events_path
- Use that on line 51 to do as-much-as-possible initialization of telemetry_writer (obv. missing activity id etc).

Then, back in CoreMain, immediately after execution_config is initialized (line 50 or so), complete the full initialization of telemetry_writer.

When assessment *or* installation runs (both should have this logic, and while today only assessment will ever run it as the execution flow diverts away in older agent case, it preserves resiliency):
- a loosely coupled check of telemetry_writer.is_agent_compatible() or somethign similar returns false -- then write appropriately to status and raise an exception.

This is a broad swathes painting above - please clarify any details that I may have skimmed over to not add even more verbosity, fill in finer details, solidify the implementation and bring back any concerns that may arise.